logging/logging.go

```diff
@@ -923,3 +928,41 @@ func printEntryToWriter(entry *logpb.LogEntry, w io.Writer) error {
 	}
 	return err
 }
+
+func (l *Logger) redirectAsJSON(pbe *logpb.LogEntry) (err error) {
+	err = serializeEntryToWriter(pbe, l.redirectOutputWriter)
+	if err == nil {
+		if internal.IngestInstrumentation {
+			internal.IngestInstrumentation = false
```

Comment:
Correct me if I wrong, but if the library is used in multithreaded environment, you might end up adding instrumentation records multiple times. Also this code is repeated below, so I would move it into separate function like `setInstrumentationStatus()` which could update the `internal.IngestInstrumentation` to `false` and return it's previous value - this way you also would be able to make a lock if it is needed.

Reply:
calling a sync.Once multiple times is low cost. I am not sure this extra logic is needed, or I am confused why it might be?

Reply:
The goal is to print instrumentation record only once per process

Reply:
I will add the locking mechanism